packages/backend/src/accounting/psql/ledger-transfer/index.ts

```diff
@@ -38,3 +72,156 @@ export async function getAccountTransfers(
+      return {
+        results: [],
+        errors
+      }
+    }
+
+    const createdTransfers = await LedgerTransfer.query(
+      trx || deps.knex
+    ).insertAndFetch(transfers.map(prepareTransfer))
+
+    return {
+      results: createdTransfers,
+      errors: []
+    }
+  } catch (error) {
+    if (error instanceof UniqueViolationError) {
+      return {
+        results: [],
+        errors: [{ index: -1, error: TransferError.TransferExists }]
+      }
+    }
+
+    const errorMessage = 'Could not create transfer(s)'
+    deps.logger.error({ errorMessage: error && error['message'] }, errorMessage)
+    throw new Error(errorMessage)
+  }
+}
+
+async function validateTransfer(
+  deps: ServiceDependencies,
+  args: ValidateTransferArgs
+): Promise<TransferError | undefined> {
+  const { amount, timeoutMs, creditAccount, debitAccount } = args
+
+  if (amount <= 0n) {
+    return TransferError.InvalidAmount
+  }
+
+  if (timeoutMs && timeoutMs <= 0n) {
+    return TransferError.InvalidTimeout
+  }
+
+  if (creditAccount.id === debitAccount.id) {
+    return TransferError.SameAccounts
+  }
+
+  if (creditAccount.ledger !== debitAccount.ledger) {
+    return TransferError.DifferentAssets
+  }
+
+  return validateBalances(deps, args)
+}
+
+async function validateBalances(
+  deps: ServiceDependencies,
+  args: ValidateTransferArgs
+): Promise<TransferError | undefined> {
+  const { amount, creditAccount, debitAccount } = args
+
+  const [creditAccountBalances, debitAccountBalance] = await Promise.all([
+    getAccountBalances(deps, creditAccount),
+    getAccountBalances(deps, debitAccount)
+  ])
+
+  if (
+    !hasEnoughDebitBalance({
+      account: creditAccount,
+      balances: creditAccountBalances,
+      transferAmount: amount
+    })
+  ) {
+    return TransferError.InsufficientDebitBalance
+  }
+
+  if (
+    !hasEnoughLiquidity({
+      account: debitAccount,
+      balances: debitAccountBalance,
+      transferAmount: amount
+    })
+  ) {
+    return TransferError.InsufficientLiquidity
+  }
+}
+
+export function hasEnoughLiquidity(args: BalanceCheckArgs): boolean {
+  const { account, balances, transferAmount } = args
+  const { creditsPosted, debitsPosted, debitsPending } = balances
+
+  return (
+    account.isSettlementAccount ||
+    creditsPosted >= debitsPosted + debitsPending + transferAmount
+  )
+}
+
+export function hasEnoughDebitBalance(args: BalanceCheckArgs): boolean {
+  const { account, balances, transferAmount } = args
+  const { creditsPosted, creditsPending, debitsPosted } = balances
+
+  return (
+    !account.isSettlementAccount ||
+    debitsPosted >= creditsPosted + creditsPending + transferAmount
+  )
+}
+
+function prepareTransfer(
+  transfer: CreateTransferArgs
+): Partial<LedgerTransfer> {
+  return {
+    amount: transfer.amount,
+    transferRef: transfer.transferRef,
+    creditAccountId: transfer.creditAccount.id,
+    debitAccountId: transfer.debitAccount.id,
+    ledger: transfer.ledger,
```

Comment:
```suggestion
    ledger: transfer.creditAccount.ledger,
```
and remove `ledger` from `CreateTransferArgs`